app/scripts/controllers/network/provider-api-tests/shared-tests.js

```diff
@@ -1108,31 +1166,144 @@ export function testsForRpcMethodSupportingBlockParam(
-                  'su',
-                ),
+      ['ETIMEDOUT', 'ECONNRESET', 'SyntaxError'].forEach(
+        (errorMessagePrefix) => {
+          if (providerType === 'infura' || errorMessagePrefix === 'ETIMEDOUT') {
+            it(`retries the request to Infura up to 5 times if an "${errorMessagePrefix}" error is thrown while making the request, returning the successful result if there is one on the 5th try`, async () => {
+              await withMockedCommunications(
+                { type: providerType },
+                async (comms) => {
+                  const request = { method };
+
+                  // The first time a block-cacheable request is made, the
+                  // block-cache middleware will request the latest block number
+                  // through the block tracker to determine the cache key. Later,
+                  // the block-ref middleware will request the latest block number
+                  // again to resolve the value of "latest", but the block number is
+                  // cached once made, so we only need to mock the request once.
+                  comms.mockNextBlockTrackerRequest({ blockNumber: '0x100' });
+                  // The block-ref middleware will make the request as specified
+                  // except that the block param is replaced with the latest block
+                  // number.
+                  //
+                  // Here we have the request fail for the first 4 tries, then
+                  // succeed on the 5th try.
+                  comms.mockRpcCall({
+                    request: buildRequestWithReplacedBlockParam(
+                      request,
+                      blockParamIndex,
+                      '0x100',
+                    ),
+                    error: `${errorMessagePrefix}: Some message`,
+                    times: 4,
+                  });
+                  comms.mockRpcCall({
+                    request: buildRequestWithReplacedBlockParam(
+                      request,
+                      blockParamIndex,
+                      '0x100',
+                    ),
+                    response: {
+                      result: 'the result',
+                      httpStatus: 200,
+                    },
+                  });
+                  const result = await withClient(
+                    { type: providerType },
+                    async ({ makeRpcCall, clock }) => {
+                      return await waitForPromiseToBeFulfilledAfterRunningAllTimers(
+                        makeRpcCall(request),
+                        clock,
+                      );
+                    },
+                  );
+
+                  expect(result).toStrictEqual('the result');
+                },
               );
             });
+          }
+
+          it(`causes a request to fail with a custom error if an "${errorMessagePrefix}" error is thrown while making the request to Infura 5 times in a row`, async () => {
```

Comment:
Another reference to Infura here that should be using `providerType`

Reply:
Fixed in [66aa9c1](https://github.com/MetaMask/metamask-extension/pull/16337/commits/66aa9c1b1c4961ba571948aea065608cb7517087).